Runtime-catalog failed to start due to migration 109

Deprecated elements (old catch and when) can contain empty priorityNumber parameter.
```
Caused by: org.flywaydb.core.internal.sqlscript.FlywaySqlScriptException: Script V109_000__fix-field-types.sql failed
-------------------------------------------
SQL State  : 22P02
Error Code : 0
Message    : ERROR: invalid input syntax for type integer: ""
Location   : db/migration/postgresql/configs/V109_000__fix-field-types.sql (/app/nested:/app/qip-runtime-catalog-0.4.1.jar!/db/migration/postgresql/configs/V109_000__fix-field-types.sql)
Line       : 2
Statement  : Run Flyway with -X option to see the actual statement causing the problem

	at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.handleException(DefaultSqlScriptExecutor.java:256)
	at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.executeStatement(DefaultSqlScriptExecutor.java:217)
	at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.execute(DefaultSqlScriptExecutor.java:137)
	at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor.executeOnce(SqlMigrationExecutor.java:75)
	at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor.lambda$execute$0(SqlMigrationExecutor.java:66)
	at org.flywaydb.core.internal.database.DefaultExecutionStrategy.execute(DefaultExecutionStrategy.java:31)
	at org.flywaydb.core.internal.resolver.sql.SqlMigrationExecutor.execute(SqlMigrationExecutor.java:65)
	at org.flywaydb.core.internal.command.DbMigrate.doMigrateGroup(DbMigrate.java:391)
	... 63 common frames omitted
Caused by: org.postgresql.util.PSQLException: ERROR: invalid input syntax for type integer: ""
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2734)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2421)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:372)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:518)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:435)
	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:357)
	at org.postgresql.jdbc.PgStatement.executeCachedSql(PgStatement.java:342)
	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:318)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:313)
	at com.zaxxer.hikari.pool.ProxyStatement.execute(ProxyStatement.java:94)
	at com.zaxxer.hikari.pool.HikariProxyStatement.execute(HikariProxyStatement.java)
	at org.flywaydb.core.internal.jdbc.JdbcTemplate.executeStatement(JdbcTemplate.java:210)
	at org.flywaydb.core.internal.sqlscript.ParsedSqlStatement.execute(ParsedSqlStatement.java:88)
	at org.flywaydb.core.internal.sqlscript.DefaultSqlScriptExecutor.executeStatement(DefaultSqlScriptExecutor.java:212)
	... 69 common frames omitted
```

--- a/src/main/java/org/qubership/integration/platform/runtime/catalog/service/exportimport/migrations/chain/V106ChainImportFileMigration.java
+++ b/src/main/java/org/qubership/integration/platform/runtime/catalog/service/exportimport/migrations/chain/V106ChainImportFileMigration.java
@@ -4,6 +4,7 @@ import com.fasterxml.jackson.databind.node.JsonNodeFactory;
 import com.fasterxml.jackson.databind.node.ObjectNode;
 import com.fasterxml.jackson.databind.node.TextNode;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
 import org.springframework.stereotype.Component;
 
 import java.util.Collection;
@@ -40,7 +41,8 @@ public class V106ChainImportFileMigration extends ChainElementsPropertiesMigrati
     private static void changePropertyTypeFromStringToInteger(ObjectNode properties, String propertyName) {
         if (properties.path(propertyName) instanceof TextNode textNode) {
             try {
-                Integer value = Integer.valueOf(textNode.asText());
+                String textValue = textNode.asText();
+                Integer value = Integer.valueOf(StringUtils.isEmpty(textValue) ? "0" : textValue);
                 properties.set(propertyName, JsonNodeFactory.instance.numberNode(value));
             } catch (NumberFormatException e) {
                 log.warn("Failed to convert {} value from string to integer", propertyName, e);

--- a/src/main/resources/db/migration/postgresql/configs/V109_001__fix-field-types.sql
+++ b/src/main/resources/db/migration/postgresql/configs/V109_001__fix-field-types.sql
@@ -1,3 +1,32 @@
+-- Changing the value of 'priorityNumber' field from empty string or null to 0 for 'when' and 'catch' elements.
+update catalog.elements e
+set
+    properties = jsonb_set(e.properties, '{priorityNumber}', to_jsonb(0))
+    from
+    catalog.chains c
+where
+    (c.id = e.chain_id)
+    and (e.type = 'when') or (e.type = 'catch')
+    and (e.properties ? 'priorityNumber')
+    and (((jsonb_typeof(e.properties->'priorityNumber') = 'string')
+            and (e.properties->>'priorityNumber' = ''))
+        or (jsonb_typeof(e.properties->'priorityNumber') = 'null'));
+
+
+-- Changing the value of 'priority' field from empty string or null to 0 for 'catch-2' element.
+update catalog.elements e
+set
+    properties = jsonb_set(e.properties, '{priority}', to_jsonb(0))
+    from
+    catalog.chains c
+where
+    (c.id = e.chain_id)
+    and (e.type = 'catch-2')
+    and (e.properties ? 'priority')
+    and (((jsonb_typeof(e.properties->'priority') = 'string')
+            and (e.properties->>'priority' = ''))
+        or (jsonb_typeof(e.properties->'priority') = 'null'));
+
 -- Changing a type of 'priorityNumber' field from string to integer for 'when' and 'catch' elements.
 update catalog.elements e
 set

